src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -55,13 +55,39 @@ public void storeMessage(Message message) {
    * @return a list of messages posted by the user, or empty list if user has never posted a
    *     message. List is sorted by time descending.
    */
-  public List<Message> getMessages(String user) {
-    return messageQuery(user);
-  }
+  public List<Message> getMessages(String recipient) {
```

Comment:
This function now looks **very** similar to the messageQuery(user), I haven't looked to much into it but do you think its possible to modify that code a little to fit the new use case??

Reply:
It would be ideal to factor out the repetitive code, but I don't know if it would be any simpler because I might need to parse the url to find out what parameter was passed in, and then do some additional case checking inside the method to set specific variables. I think it might be cleaner to leave as is for now?

Reply:
I think if set up correctly it could be clean, though there may be case checking, there is in any nice API surface that is created. I've always found that the repeating of code in different methods can confuse me more than the if or switch statements that come in a configurable API.

I was thinking the parameters would be along the lines of the following:
`private List<Message> messageQuery(String filterType, String filterValue)`

From these you should be able to determine the way the new Messages should be created. If you have any questions definitely reach out by chat, its a little easier to correspond there :)

Reply:
combined methods, now updated